opensquirrel/ir.py

```diff
@@ -309,7 +314,10 @@ def identity(q: Qubit) -> BlochSphereRotation:
         return BlochSphereRotation(qubit=q, axis=(1, 0, 0), angle=0, phase=0)
 
     def __repr__(self) -> str:
-        return f"BlochSphereRotation({self.qubit}, axis={self.axis}, angle={self.angle}, phase={self.phase})"
+        axis = np.round(self.axis, self._significant_digits_repr)
+        angle = np.round(self.angle, self._significant_digits_repr)
+        phase = np.round(self.phase, self._significant_digits_repr)
```

Comment:
I'd recommend to have a `_round(f: float): float` function that would just `return np.round(f, self._significant_digits_repr)`, then rewrite all the calls to `np.round(blah, self._significant_digits_repr)` as `self._round(blah)`.

Reply:
Could point! Implemented 😄